apps/src/weblab/brambleHost.js

```diff
@@ -52,28 +52,58 @@ function putFilesInBramble(sources, callback) {
     // create project root directory
     sh.mkdirp(projectRoot, err => {
 
-      function writeFileData(path, data, callback) {
-        path = Path.join(projectRoot, path);
+      function writeFileDataAndContinue(filename, data, currentIndex) {
+        var path = Path.join(projectRoot, filename);
         // write the data
-        fs.writeFile(path, data, err => {
-          // call completion callback
-          callback(err);
-        });
-      }
-
-      // async-chained enumeration: write the i-th file into Bramble file system
-      function writeSourceFile(sources, i, callback) {
-        if (i < sources.files.length) {
-          const file = sources.files[i];
-          // write file data into Bramble
-          writeFileData(file.name, file.data, () => {
+        function onWriteComplete(err) {
+          if (err) {
+            // call completion callback
+            callback(err);
```

Comment:
This error handler doesn't make sense anymore, since this function no longer takes a `callback`.